pkg/githubsource/receive_adapter.go

```diff
@@ -61,16 +68,20 @@ func (ra *GitHubReceiveAdapter) handleEvent(payload interface{}, gitHubEventType
 	cloudEventType := fmt.Sprintf("%s.%s", sourcesv1alpha1.GitHubSourceEventPrefix, gitHubEventType)
 	source := sourceFromGitHubEvent(gh.Event(gitHubEventType), payload)
 
-	return ra.postMessage(payload, source, cloudEventType, eventID)
+	return ra.postMessage(payload, source, cloudEventType, eventID, hdr)
 }
 
-func (ra *GitHubReceiveAdapter) postMessage(payload interface{}, source, eventType, eventID string) error {
+func (ra *GitHubReceiveAdapter) postMessage(payload interface{}, source, eventType, eventID string, hdr http.Header) error {
 	ctx := cloudevents.EventContext{
 		CloudEventsVersion: cloudevents.CloudEventsVersion,
 		EventType:          eventType,
 		EventID:            eventID,
 		EventTime:          time.Now(),
 		Source:             source,
+		Extensions: map[string]interface{}{
```

Comment:
I'm guessing that this is the key bit?  Thanks for the fix!

Reply:
Yeah, my understanding of the cloud event specs is that Extensions is the right place to propagate this kind of metadata. Although, note that they mandate CE-X as the prefix for the keys, this means that the consumer downstream will have to s/CE-// to get back the headers wanted by the GH sdk.